### PR TITLE
Created test for ByteBuffer + Improved TextTest

### PR DESCRIPTION
**Pull request by marcusnaslund**
Created a simple test for `ByteBuffer` to replace the non-existant previous one.
Also added missing tests for `Text`.

### CODE REVIEW

File: test/base/ByteBufferTest.ooc [hunk 1]
```diff
@@ -1,3 +1,48 @@
...
+			for (i in 0 .. 1024 / 8)
+				expect(buffercopy pointer[i] as Int, is equal to(buffer pointer[i] as Int))
+			buffercopy free()
+		})
+		this add("slice", func {
+			buffer := ByteBuffer new(1024)
+			for (i in 0 .. 1024 / 8)
+				buffer pointer[i] = i
+			slice := buffer slice(10, 8)
+			buffer free()
```
Comment (davidhesselbom) [inline test/base/ByteBufferTest.ooc]: This looks like a bad idea - `slice` still refers to the same memory as `buffer` does. Try `buffer referenceCount decrease()` instead.

Comment (marcusnaslund) [inline test/base/ByteBufferTest.ooc]: You're completely right. Fixed.

File: test/base/ByteBufferTest.ooc [hunk 2]
```diff
@@ -1,3 +1,48 @@
...
+	init: func {
+		super("ByteBuffer")
+		this add("set data", func {
+			buffer := ByteBuffer new(1024)
+			expect(buffer size, is equal to(1024))
+			for (i in 0 .. 1024 / 8)
+				buffer pointer[i] = i
+			for (i in 0 .. 1024 / 8)
+				expect(buffer pointer[i] as Int, is equal to(i))
+			buffer free()
```
Comment (davidhesselbom) [inline test/base/ByteBufferTest.ooc]: Always do `referenceCount decrease()` on things that have a refcounter, don't free them like this.

**Comment by marcusnaslund** [comment]
Updated: `ByteBufferTest` now updates the refcounter instead.

**Comment by davidhesselbom** [comment]
Thanks.

File: test/base/TextTest.ooc [hunk 3]
```diff
@@ -148,9 +148,24 @@ TextTest: class extends Fixture {
...
 			expect(Text new("6.5E5") toFloat(), is equal to(6.5f * pow(10, 5) as Float) within(tolerance))
 			expect(Text new("-34.5E-2") toFloat(), is equal to(-0.345f) within(tolerance))
 		})
+		this add("MakeTextLiteral", func {
+			mytext := makeTextLiteral(c"Hello", 5)
+			mytext2 := t"Hello"
+			expect(mytext == mytext2)
+		})
+		this add("copyTo", func {
+			t := Text new("test string")
```
Comment (davidhesselbom) [inline test/base/TextTest.ooc]: Call `t` and `t2` `text` and `text2` or something.

Comment (marcusnaslund) [inline test/base/TextTest.ooc]: Done.

**Pull request merged**